diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/AvoidEmptyPrimaryConstructor.kt

```diff
@@ -26,14 +29,14 @@ class AvoidEmptyPrimaryConstructor(private val configRules: List<RulesConfig>) :
 
     @Suppress("UnsafeCallOnNullableType")
     private fun checkCLass(node: KtClass) {
-        if(node.primaryConstructor?.valueParameters?.isNotEmpty() != false)
+        println(node.node.prettyPrint())
+        if(node.primaryConstructor?.valueParameters?.isNotEmpty() != false || node.primaryConstructor?.isPrivate() == true)
```

Comment:
Primary constructor can also be annotated, you should check if it has any modifiers at all.